### PR TITLE
Xenobiology Vatgrowing: From microbiological sample to full grown mob

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

This pull request adds in vatgrowing, a task which xenobiologist players will have to do along-side of their usual slime related bussinesses.

Vat-growing takes a step back and focuses more on the cultivating of mobs rather than just focusing on slimes. You start by looking for samples on surfaces (with a swab) or on specific mobs (with a biopsy tool). The idea is that this makes you go look for unique samples in maint on dirty things such as dumpsters and hobo beddings (to be added). Once you find a sample it will consist of multiple micro-organisms. These micro-organisms are from a specific mob, but the details are vague (More experienced players would recognize it, ofcourse). Using science goggles you can see some more details, such as which reagents are required to: Grow it, help it grow faster, and stop it from growing as fast.

These details are important, as in our next section we will use the new "growing vat" to go from micro-organism, to full-blown creature. This requires some work from the player however. First of all, all creatures have a list of reagents it requires to grow at all.  Lets set up an example case. We went to maint and find a sample, it has the micro-organisms of a mouse, cockroach and chicken.

Mice organisms need nutriment and water. Like watermelon juice, and hate corn oil
Chicken organisms need nutriment, like water, and hate oil.
Cockroach organisms need cooking oil, like vitamins, and hate synthflesh.
(these likes/dislikes will change after a balance pass from Kryson)

Now, in our case we'd like to grow some mice for some reason. We fill the tank up with nutriment and water as these are required to even have the organisms grow. And adding some watermelon juice will help the mouse organism grow even faster, but isn't required. We avoid adding corn oil because it will slow growth. However, we might add oil. Because the chicken organism will grow due to there being nutriments in the vat.

Why care about this? Because you only want one micro-organism to grow. If you manage to fully grow a single organism from a sample, but another one in the same sample is grown too far, you risk fucking up and destroying the sample (And maybe cause other unintended effects)

There are some smaller minor details, such as viruses which can be found in samples which might affect some micro-organisms more than others. But that is still being fleshed out a bit.

In a future PR, the idea is also to adjust slime feeding needs to require xenobiologist players to go out and find new samples. And remove some of the mobs of the gold cores so xenobiologist players don't have a simple meta to go for.

A video can be found below of the above scenario playing out successfully.
https://streamable.com/5dc1a

## Why It's Good For The Game

This brings xeno-biology somewhat more towards an experimentation and research direction, focusing on trying to create different mobs. This system also invokes xenobiologists to leave their departments and find samples in weird spots. Most likely at first these spots will mostly be in maint (I used a disposal bin as an example though)

This system will also be used to provide food for slimes later on, as this allows us to make it so specific slimes might need specific mobs to grow. (Green slimes eating only chickens etc)

Kryson did most of the design for this PR, while I focused on code. Further balance will mostly be done by Kryson while I'll ensure it gets into this PR.

- [x] Add audio for everything
- [ ] Add more content (New mobs)
- [ ] Add full virus functionality (Essentialy, the ability to get a virus micro_organism and turn it into a real virus
- [ ] Add the swabbable element to spots where it would make sense to have (Maint items etc)
- [ ] Ensure I autodoc everything
- [ ] Find a better way to display information on likes/dislikes of micro organisms (Microscope most likely)

Slime refactor will mostly happen in a seperate PR, as this PR will get too big otherwise.

## Changelog
:cl: Kryson and Qustinnus
add: Adds vat-growing for xenobiology.  Go to maint, find some samples, and grow some cool mobs! More info on the PR https://github.com/tgstation/tgstation/pull/49581
balance: Adds protein and rebalances a lot of meat items with it
/:cl:

### CODE REVIEW

**Comment by Qustinnus** [comment]
and i'd like to emphasis; most of the design is from @Krysonism so be sure to hail him for it :)

**Comment by Qustinnus** [comment]
Also I need to look at how fix the linter problem, I don't recognize the issue but I need a global list to ensure swabbable can be an element so I'll look into that tommorow.

File: code/datums/components/swabbing.dm [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	if(!do_after(user, 30, TRUE, target)) // Start swabbing boi
+		return
+
+	if(!SEND_SIGNAL(target, COMSIG_SWAB_FOR_SAMPLES, swabbed_items)) //If we found something to swab now we let the swabbed thing handle what it would do, we just sit back and relax now.
+		to_chat(user, "<span class='warning'>You do not manage to find a anything on [target]!</span>")
+		return
+
+	to_chat(user, "<span class='nicegreen'>You manage to collect a microbiological sample from [target]!</span>")
+
+	target.RemoveElement(/datum/element/swabable)
```
Comment (Qustinnus) [inline code/datums/components/swabbing.dm]: This doesn't seem to work @ninjanomnom do you know why?

**Comment by PKPenguin321** [comment]
So it's like gooncloning but with every mob except humans? Pretty neat

**Comment by Jessen21** [comment]
as good an idea it is, we should make sure it also doesn't just block the people who want to continue as it is. If you're going to make it where certain mobs are better for certain slimes, thats good, but we shouldn't disable them from just working with monkies too. Just have monkies as like a base, and different mobs feed the slimes more efficiently.

**Comment by Qustinnus** [comment]
@Jessen21 Could you give me somearguments as to why that's important? This broadens the gameplay loop for xenobiology so its not just waiting for slimes to suck on monkeys, and instead makes you have to go through more work to get their food. (I'd also raisen the grow speed for slimes if they have food so its no longer a wait game)

**Comment by Krysonism** [comment]
> 
> 
> as good an idea it is, we should make sure it also doesn't just block the people who want to continue as it is. If you're going to make it where certain mobs are better for certain slimes, thats good, but we shouldn't disable them from just working with monkies too. Just have monkies as like a base, and different mobs feed the slimes more efficiently.

Oranges has requested we diversify slime diets

The food the slimes require will get more complex the higher the tier. 

Lower tier slimes might eat mobs that are able to produced by non-vat methods such as chickens and mice and the orange slimes might only eat husked monkeys and cooked steaks.

We want to increase active work such as food preparation, building stuff and gathering resources while decreasing the time spent waiting.

I want xenobio to be about taking care of your creatures, not standing around waiting to receive gamer gear.

**Comment by Jessen21** [comment]
I main xenobiology most of the time. I can tell you that I'd at least like something to remain as it is, because there are far too many xenobiology nerfs or changes going on lately. If you must diversify the diets, I get that, but at least have something that will get the job done at a slower rate but still doable for people who don't want to spend half the round trying to just find a specific mob to duplicate and cost MORE time to do just for a slightly quicker speed at producing. I can tell you point blank that most of the time Xenobiologists can't even get started on their work because something hazardous to the station has begun by the time they've even got their equipment set up proper, gotten the materials they need, ect.

**Comment by Shadowflame909** [comment]
Tfw you gotta cultivate dragon eggs/xeno eggs 1 hour in to feed those rainbow slimes.

In all honesty this is gonna be a time-gate.

But, could be a cool time gate.

Cool idea would be having better food sources, replace the slime recycler entirely, So something powerful would make the slime split into more.

**Comment by Jessen21** [comment]
> 
> 
> Tfw you gotta cultivate dragon eggs/xeno eggs 1 hour in to feed those rainbow slimes.
> 
> In all honesty this is gonna be a time-gate.
> 

exactly. and most rounds not even MAKING it to an hour, makes it an unneeded time gate.

If you really want to give it a benefit of faster speed on certain food types, thats fine, but if you're just going to COMPLETELY lock them behind specific food types, then people are just going to stop doing Xenobiology all together and all this work would just be a waste.

**Comment by ATH1909** [comment]
This PR is cool,

> In a future PR, the idea is also to adjust slime feeding needs to require xenobiologist players to go out and find new samples. And remove some of the mobs of the gold cores so xenobiologist players don't have a simple meta to go for.

but this is cringe.

Xenobiologist is already a job that struggles with getting its cool toys before the shuttle's been called, and you want to make it SLOWER? Even if you made slime evolution INSTANT with the right kinds of food, making specific kinds of food/mobs a REQUIREMENT to grow certain species of slimes will slow down xenobiology so much that it'll be a miracle if the xenobiologists even manage to get yellow slime core cells for the cyborgs before the round ends.

**Comment by Jessen21** [comment]
> 
> 
> This PR is cool,
> 
> > In a future PR, the idea is also to adjust slime feeding needs to require xenobiologist players to go out and find new samples. And remove some of the mobs of the gold cores so xenobiologist players don't have a simple meta to go for.
> 
> but this is cringe.
> 
> Xenobiologist is already a job that struggles with getting its cool toys before the shuttle's been called, and you want to make it SLOWER? Even if you made slime evolution INSTANT with the right kinds of food, making specific kinds of food/mobs a REQUIREMENT to grow certain species of slimes will slow down xenobiology so much that it'll be a miracle if the xenobiologists even manage to get yellow slime core cells for the cyborgs before the round ends.

EXACTLY. you'd just be tying hands even more then already done. You guys have practically nerfed everything about Xenobiology that made it very viable or fun without being too bat guano crazy. I mean, theres no longer xenobio changelings, Magicarps are getting removed... 

if you want my 2 cents on it. Maybe these mobs should be used to guarantee at least ONE split of the color that likes them from it's previous tier color, allowing a quicker experience from there and giving it use, but not damaging the CORE functionality of the Xenobiology system so bad that no one wants to do it anymore.

**Comment by Qustinnus** [comment]
this pr doesn't actually time gate it to an hour so maybe save this discussion for when that actually happens is best.

Also, oranges literaly said jobs should be an hour to 90 minutes

**Comment by Shadowflame909** [comment]
My thoughts on this PR are simple Quint.

Rework the goal of vatgrowing.

You could use this concept and have it replace the slime recycler, or at least the "creates more slime cores" part of it.

Imagine a /tg/ station where the xenobiologist could use regular monkeys, and they'd only get like 1 slime core from every slime. Or, they instead go across the station scavenging, and create Gorillas. Upon which the slimes eat, causing them to drop two slime cores upon recycling.

This would effectively create a system that would promote how fast you get the toys you want, with how much you interact with the other stations inhabitants.

That would be great.

Like I said with hygiene though. Negatively reinforcement to cause interaction will only make people bitter and salt at you. If you have a base-line, and instead make that interaction optional, but have increased improvements for your slime farming. Then you're gonna get love, and reach the goal that you want.

TLDR: Should replace the slime recycler for more slime cores. Not the monkeys.

**Comment by Jessen21** [comment]
> 
> 
> this pr doesn't actually time gate it to an hour so maybe save this discussion for when that actually happens is best.
> 
> Also, oranges literaly said jobs should be an hour to 90 minutes

oh no, this is ENTIRELY a time gate. Theres no getting around that. If you have to go swabbing for mobs, leaving your lab alone for who knows what person wants to break in and do it themselves disrupting your work, then its time consuming and going to overall reduce the desire to do it more. Jobs shouln't take an hour to an hour and a half to get a result. Anyone who says otherwise hasn't been playing the game nearly to the level most people who have been doing that job would have.

**Comment by Spepbuthewranglesgithub** [comment]
> The idea is that this makes you go look for unique samples in maint on dirty things such as dumpsters and hobo beddings (to be added). 
> maint
> In a future PR, the idea is also to adjust slime feeding needs to require xenobiologist players to go out and find new samples. And remove some of the mobs of the gold cores so xenobiologist players don't have a simple meta to go for.
> 
> ## Why It's Good For The Game
> 
> This system will also be used to provide food for slimes later on, as this allows us to make it so specific slimes might need specific mobs to grow. (Green slimes eating only chickens etc)

better give scientists maint access before the slime PR or your existance privileges are being revoked.

_at the same time the creatures are either going to be completely boring, like chickens, or more gold slime tier lads that will swarm over the station in droves as some idiot makes 20 vats and starts producing an army of watchers_

**Comment by Qustinnus** [comment]
This isn't negative reinforcement, I'm not actively telling you off for not doing X. X is just a requirement. Negative reinforcement is when I have you killed for trying to feed a black slime the wrong food.

Either way, take it up to oranges and kryson as I think this is fine

**Comment by Jessen21** [comment]
> 
> 
> > The idea is that this makes you go look for unique samples in maint on dirty things such as dumpsters and hobo beddings (to be added).
> > maint
> > In a future PR, the idea is also to adjust slime feeding needs to require xenobiologist players to go out and find new samples. And remove some of the mobs of the gold cores so xenobiologist players don't have a simple meta to go for.
> > ## Why It's Good For The Game
> > This system will also be used to provide food for slimes later on, as this allows us to make it so specific slimes might need specific mobs to grow. (Green slimes eating only chickens etc)
> 
> better give scientists maint access before the slime PR or your existance privileges are being revoked.
> 
> _at the same time the creatures are either going to be completely boring, like chickens, or more gold slime tier lads that will swarm over the station in droves as some idiot makes 20 vats and starts producing an army of watchers_

oh yeah, i'd just use the watcher methood instead of the gold slimes at that point. I've been TRYING to tell people that removing them from gold slimes was a mistake.

**Comment by Qustinnus** [comment]
Either way, people always complain that jobs don't have content for 90 minutes so the entire point is that jobs lengthen up their full content to 90 minutes, with 60 minutes being the point you reached a lot of core things. This is the benchmark and if you dislike that you should instead be argueing about oranges because that's the design direction that's been set up for tg for like a year now

**Comment by Jessen21** [comment]
I certainly haven't been complaining. I felt Xenobiology was lengthy enough as it was. you guys are just seeming to be on a trip of trying to nerf it do death so that no one wants it. What, do you guys want to replace it with something else?

**Comment by Qustinnus** [comment]
yes

**Comment by Spepbuthewranglesgithub** [comment]
> I certainly haven't been complaining. I felt Xenobiology was lengthy enough as it was. you guys are just seeming to be on a trip of trying to nerf it to death so that no one wants it. What, do you guys want to replace it with something else?



> yes


but what if, no?

**Comment by XDTM** [comment]
Nerfs aside xenobiology already has the highest amount of content of all the jobs _by far_. I'm not saying that the PR is necessarily bad, but xenobio should probably stop getting new features so other departments can be developed a bit as well.

File: code/datums/components/swabbing.dm (hunk 1 above)
Comment (Time-Green) [reply code/datums/components/swabbing.dm]: You need the exact same parameters as when you added it

**Comment by Shadowflame909** [comment]
> 
> 
> Nerfs aside xenobiology already has the highest amount of content of all the jobs _by far_. I'm not saying that the PR is necessarily bad, but xenobio should probably stop getting new features so other departments can be developed a bit as well.

Maybe after I rework that staff of change removal

**Comment by Tlaltecuhtli** [comment]
so whats the endgame? giving microbes banana juice until they turn into gorillas and similar things?

**Comment by GuyonBroadway** [comment]
The problem is that even if a job takes 90 mins rounds seldom last more than 40 and rounds will never be longer unless you prevent calling the shuttle for an hour. Captains will always call the shuttle at half an hour because "shits fukt" 

Also when people ask for things to do beyond 90 mins they don't mean "take what we could already do and add a waiting period" they mean "give the job something that can meaningfully contribute to the round 90 mins in".

**Comment by Krysonism** [comment]
> Nerfs aside xenobiology already has the highest amount of content of all the jobs _by far_. I'm not saying that the PR is necessarily bad, but xenobio should probably stop getting new features so other departments can be developed a bit as well.

It has the highest amount of gamer gear for sure, but I think xenobio is actually in a bad place since the actual job gameplay is boring, tedious and frustrating. 

The simple mob / sentience interaction is by far the most interesting and thematically appropriate part of xenobio.

In addition this helps solve some of my biggest gripes with xenobio, you often work alone since having more workers do not really increase productivity and you never leave your little isolated corner of the station because xenobio is almost completely self sufficient.

I don't want to do boring repetitive shit atop a mountain of bombs, guns and healing items. I want to feel like a mix of Dr. Doolittle and Shou Tucker.

**Comment by FantasticFwoosh** [comment]
That's a pity, i like the swabbing/sample collection aspect of it but i thought you were going to explore the origin of monkey cubes, for a few home-brewed self made creations like a cow-cube or a bear-cube made out of exotic meat samples.

Instead while i appreciate the effort took to giving it depth it seem just like a novelty gold-extract with more process, the smoke particularly looks very annoying and not much reason to maintain a chem lab/chem stock in itself.

**Comment by Qustinnus** [comment]
@FantasticFwoosh the smoke was just temporary so I knew when it was done. and the idea is we can add self-made creatures later easily. I'm not going to add everything in one PR as that'll kill the PR

**Comment by Qustinnus** [comment]
@GuyonBroadway I can't tell if you're serious or not but removing the need to wait on slimes by making them eat/grow faster but require specific food which you have to get through an active task (by getting samples and growing them) does not seem like something that requires you to wait more. If anything you wait less, do more.

**Comment by trollbreeder** [comment]
Why not horrifying clown mutants if you fuck up the organism?

**Comment by Qustinnus** [comment]
The new mobs are still in the works

File: code/_globalvars/lists/xenobiology.dm [hunk 2]
```diff
@@ -0,0 +1,14 @@
+///These global lists exist to allow our element to have weight tables without having to be seperate instances.
+
+///Assoc list of cell line define | assoc list of datum | cell_line
+GLOBAL_LIST_INIT(cell_line_tables, list(
```
Comment (spookydonut) [inline code/_globalvars/lists/xenobiology.dm]: GLOBAL_LIST_INIT_TYPED

Comment (spookydonut) [reply code/_globalvars/lists/xenobiology.dm]: needs to be /list/list/

**Comment by GuyonBroadway** [comment]
@Questinnus 

It was more in response to the "oranges wants jobs to take 90 mins/people want to have stuff to do after 90 mins" comments.

I like the idea of, as people mentioned, certain slimes growing faster off of some mobs and slower off others. Perhaps confer a bonus chance to mutate into a prefered type if you feed a slime that types favoured food (which may be distasteful to the current breed)  thus rewarding productivity and some time with reduced rng. And less rng is better

**Comment by plapatin** [comment]
i will support this if you allow me a way to cultivate glockroaches

**Comment by Ty-the-Smonk** [comment]
It’s almost like, xenobio, has a slot for 2 people, and doing slimes with 2 people sucks

**Comment by FantasticFwoosh** [comment]
>        i will support this if you allow me a way to cultivate glockroaches

Just add more sulphur & mutagen to the cockroach cultures, though it would be more deep if it was a combined effort of a lengthy growing process as a result, because free sentient glockroaches for all _might possibly_ not be fun for a slow-burn xenobiology monster flood.

File: code/_globalvars/lists/xenobiology.dm (hunk 2 above)
Comment (Qustinnus) [reply code/_globalvars/lists/xenobiology.dm]: wtf i thought i already made this init_typed, odd

**Comment by IndieanaJones** [comment]
> 
> 
> > ```
> >    i will support this if you allow me a way to cultivate glockroaches
> > ```
> 
> Just add more sulphur & mutagen to the cockroach cultures, though it would be more deep if it was a combined effort of a lengthy growing process as a result, because free sentient glockroaches for all _might possibly_ not be fun for a slow-burn xenobiology monster flood.

Well, it's either that or you spam blobbernauts, so I guess you pick one.

**Comment by Iatots** [comment]
Swabbing organisms directly should give you a pure sample, which you could then fuck with mutagens and other stuff.

**Comment by FantasticFwoosh** [comment]
> 
> 
> Swabbing organisms directly should give you a pure sample, which you could then fuck with mutagens and other stuff.

Id even go as far as to say to replace the function of gold core slimes entirely to just procure random DNA samples to analyze, cultivate and manage and greatly increase the value of the 'life' chem reagent.

**Comment by Qustinnus** [comment]
WYCI

**Comment by AdamElTablawy** [comment]
seems like it'd work fine for manuel

**Comment by optimumtact** [comment]
Arguing about the xenobio nerfs on this PR is offtopic.

If you want to make a case, you should hit me up, but right now I am expecting a significant raise in slime care complexity.

**Comment by Krysonism** [comment]
> 
> 
> Swabbing organisms directly should give you a pure sample, which you could then fuck with mutagens and other stuff.

This is what the biopsy tool does.

biopsy_comp

**Comment by Krysonism** [comment]
> so whats the endgame? giving microbes banana juice until they turn into gorillas and similar things?

We have two new thematically appropriate creatures in development that I expect to be pretty good.

Stat boosted "rare" versions of the growable mobs are also planned.

In the future I would also like to see chimaeric hybrid creatures, although this will require a bit of sprite work.

I would also like to see most helpful mobs like the lightgeist.

**Comment by Krysonism** [comment]
Okay so the main resource of vat growing is protein. The main reason for this is to create a way to recycle surplus mobs back into the process.

Protein can now be found in meat, soy and to a lesser extent cheese and eggs. Protein works a lot like nutriment in that it slowly heals brute. 

To somewhat counterbalance this, nutriment now heals slightly less. What this means in the end is that you will heal a bit more, and a bit faster if you eat a balanced diet that contains both nutriment and protein, but a bit less if you eat foods that only contain nutriment(which in my mind represents starches).

I will try to limit the use of dispenser reagents in cell line recipes to prevent a botany type situation where building a chem dispenser becomes deeply ingrained into the meta.

My goal is that the basic creatures will be pretty easy to grow but the strongest creatures will require a bit of a scavenger hunt.

This is a bit of a challenge since a lot of reagents can be dispensed by some sort of dispenser or are completely trivial to obtain  from some other source, but i think i can pull it off.

**Comment by Krysonism** [comment]
Okay,
the main resource of vat growing is protein; a new reagent found in meat, soy and to a lesser extent cheese and eggs. The main reason for this is to create a way to recycle surplus mobs back into the process.

Protein works a lot like nutriment in that it slowly heals brute. 

To somewhat counterbalance this, nutriment now heals slightly less. What this means in the end is that you will heal a bit more, and a bit faster if you eat a balanced diet that contains both nutriment and protein, but a bit less if you eat foods that only contain nutriment(which in my mind represents starches).

I will try to limit the use of dispenser reagents in cell line recipes to prevent a botany type situation where building a chem dispenser becomes deeply ingrained into the meta.

My goal is that the basic creatures will be pretty easy to grow but the strongest creatures will require a bit of a scavenger hunt.

This is a bit of a challenge since a lot of reagents can be dispensed by some sort of dispenser or are completely trivial to obtain  from some other source, but i think i can pull it off.

**Comment by RaveRadbury** [comment]
Currently we have Vitamin, Nutriment, and Sugar. After a fashion these relate to real-world nutrition as fat, protein, and carbohydrate, respectively.

I'm all for seeing us move away from the old system to a new one that's more representative of actual nutrition concepts. Just replace nutriment with protein, vitamin with fat and sugar with carbohydrate and it would clean up a lot of stuff. This might be out of scope, but it's something worth thinking about.

**Comment by Tlaltecuhtli** [comment]
can i just farm silver slimes once and the grind all the food they spawn?

**Comment by Krysonism** [comment]
> can i just farm silver slimes once and the grind all the food they spawn?

That will get you pretty far I would imagine. I have pretty strong dislike for sliver slimes since they not only completely invalidate the chef, they also give the xenobiologist 357. revolvers and power crêpes.

However I don't know what to do about it, so no changes in that regard are planned at the moment.

**Comment by Iatots** [comment]
> 
> Currently we have Vitamin, Nutriment, and Sugar. After a fashion these relate to real-world nutrition as fat, protein, and carbohydrate, respectively.

Some fatty foods can contain certain vitamins because that's where they are stored, but I would call it a stretch to equate vitamins to a macronutrient. I'd rather have fat and oil be their own separate thing.

**Comment by Tlaltecuhtli** [comment]
will it be ""dynamic"" like you have a bunch of starting strains things but their results can change depending on whats being fed and how the chems are being balanced like on a common strain if you fed it mostly milk it turns into mouse or if you fed it plant nutriment it turns into tomato mob with rare chance of being a living tree

**Comment by Qustinnus** [comment]
yes and no, you get a sample which can have different cell line, but the cell lines in a sample could always be different.

**Comment by Krysonism** [comment]
ActionButtonComp64

Work is still ongoing behind the scenes. Here is a little sneak peak for all the good people of github.

**Comment by moocowswag** [comment]
As someone who does xenobio alot, i love it along with planned changes.

1 It gives scientists a reason to go into maint, antag scientists that regularly go in maint are normally suspicious so antag scientist isnt as stressful, maint is scary and scientists already have ways to prepare for scary.

1.5 now scientists have a reason to turn themselves into shadow people and dominate maint.

2. Now more people can work in xenobio and if taken further multiple people can work to speed it up by doing multiple jobs, im looking forward to the next pr.

**Comment by Qustinnus** [comment]
You have to keep sweeping either way because you cant keep getting samples

File: code/_globalvars/lists/xenobiology.dm [hunk 3]
```diff
@@ -0,0 +1,14 @@
+///These global lists exist to allow our element to have weight tables without having to be seperate instances.
+
+///Assoc list of cell line define | assoc list of datum | cell_line
+GLOBAL_LIST_INIT(cell_line_tables, /list/list(
```
Comment (Rohesie) [inline code/_globalvars/lists/xenobiology.dm]:
```suggestion
GLOBAL_LIST_INIT(cell_line_tables, list(

```

Comment (Qustinnus) [reply code/_globalvars/lists/xenobiology.dm]: ok i will commit this but it'll throw a liner error that I couldnt fix :^(

File: code/__DEFINES/dcs/signals.dm [hunk 4]
```diff
@@ -405,7 +405,9 @@
 #define COMSIG_TRY_STORAGE_QUICK_EMPTY "storage_quick_empty"			//(loc) - returns bool - if loc is null it will dump at parent location.
 #define COMSIG_TRY_STORAGE_RETURN_INVENTORY "storage_return_inventory"	//(list/list_to_inject_results_into, recursively_search_inside_storages = TRUE)
 #define COMSIG_TRY_STORAGE_CAN_INSERT "storage_can_equip"				//(obj/item/insertion_candidate, mob/user, silent) - returns bool
-
+// /datum/component/swabbing signals
+#define COMSIG_SWAB_FOR_SAMPLES "swab_for_samples"						//()
+	#define COMPONENT_SWAB_FOUND 1
```
Comment (Rohesie) [inline code/__DEFINES/dcs/signals.dm]:
```suggestion
	#define COMPONENT_SWAB_FOUND (1<<0)
```
Just to clarify it's a bitflag at its core.

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 5]
```diff
@@ -327,7 +327,7 @@
 			FindTarget()
 
 
-/mob/living/simple_animal/hostile/proc/AttackingTarget()
+/mob/living/simple_animal/hostile/proc/AttackingTarget(var/atom/A)
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]:
```suggestion
/mob/living/simple_animal/hostile/proc/AttackingTarget(atom/A)
```

File: code/modules/mob/living/simple_animal/hostile/ooze.dm [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+/mob/living/simple_animal/hostile/ooze/Initialize()
+	. = ..()
+	create_reagents(300)
+
+/mob/living/simple_animal/hostile/ooze/attacked_by(obj/item/I, mob/living/user)
+	if(!check_edible(I))
+		return ..()
+	eat_atom(I)
+
+/mob/living/simple_animal/hostile/ooze/AttackingTarget(var/atom/A)
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/ooze.dm]:
```suggestion
/mob/living/simple_animal/hostile/ooze/AttackingTarget(atom/A)
```

File: code/modules/mob/living/simple_animal/hostile/ooze.dm [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+			nutrition_change += consumption_amount * consumable.nutriment_factor
+		reagents.remove_reagent(R.type, consumption_amount)
+	nutrition_change -= 0.15
+	adjust_ooze_nutrition(nutrition_change)
+
+	if(ooze_nutrition <= 0)
+		adjustBruteLoss(0.5)
+
+///Returns whether or not the supplied movable atom is edible.
+/mob/living/simple_animal/hostile/ooze/proc/check_edible(var/atom/movable/AM)
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/ooze.dm]:
```suggestion
/mob/living/simple_animal/hostile/ooze/proc/check_edible(atom/movable/AM)
```

File: code/datums/components/swabbing.dm [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+	if(swabbed_items.len >= 3)
+		to_chat(user, "<span class='warning'>You cannot collect another sample on [parent]!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You start swabbing [target] for samples!</span>")
+	if(!do_after(user, 30, TRUE, target)) // Start swabbing boi
+		return
+
+	if(!SEND_SIGNAL(target, COMSIG_SWAB_FOR_SAMPLES, swabbed_items)) //If we found something to swab now we let the swabbed thing handle what it would do, we just sit back and relax now.
+		to_chat(user, "<span class='warning'>You do not manage to find a anything on [target]!</span>")
```
Comment (trollbreeder) [inline code/datums/components/swabbing.dm]:
```suggestion
		to_chat(user, "<span class='warning'>You do not manage to find anything on [target]!</span>")
```

File: code/modules/research/xenobiology/vatgrowing/samples/_microorganism.dm [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+			risk += cell_line.growth * 0.6 //60% per cell_line potentially. Kryson should probably tweak this
+	playsound(vat, 'sound/effects/splat.ogg', 50, TRUE)
+	if(rand(1, 100) < risk) //Fail roll!
+		fuck_up_growing(vat)
+		return FALSE
+	succeed_growing(vat)
+	return TRUE
+
+/datum/micro_organism/cell_line/proc/fuck_up_growing(var/obj/machinery/plumbing/growing_vat/vat)
+	vat.visible_message("<span class='warning'>the biological sample in [vat] seems to have dissipated!</span>")
```
Comment (trollbreeder) [inline code/modules/research/xenobiology/vatgrowing/samples/_microorganism.dm]: Forgot to uppercase first letter
```suggestion
	vat.visible_message("<span class='warning'>The biological sample in [vat] seems to have dissipated!</span>")
```

**Comment by Qustinnus** [comment]
still working on this fyi, corona virus (and half life alyx) has been messing with my serious PR coding time. Still working on the new mobs.

**Comment by optimumtact** [comment]
asdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdf

**Comment by Krysonism** [comment]
I've been sick and not able to do as much as I would have liked but I am feeling better and things are moving in the right direction.

**Comment by Qustinnus** [comment]
gonna re-open this because github fucked the fork this is on
